engine/cdn/objectstore/objectstore.go

```diff
@@ -0,0 +1,139 @@
+package objectstore
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/ovh/cds/sdk"
+)
+
+// Driver allows artifact to be stored and retrieve the same way to any backend
+// - Openstack / Swift
+// - Filesystem
+type Driver interface {
+	GetProjectIntegration() sdk.ProjectIntegration
+	Status() sdk.MonitoringStatusLine
+	Store(o Object, data io.ReadCloser) (string, error)
+	ServeStaticFiles(o Object, entrypoint string, data io.ReadCloser) (string, error)
+	Fetch(o Object) (io.ReadCloser, error)
+	Delete(o Object) error
+	TemporaryURLSupported() bool
+}
+
+// DriverWithRedirect has to be implemented if your storage backend supports temp url
+type DriverWithRedirect interface {
```

Comment:
I think the temp URL are useless with this new service

Reply:
To delete.